confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/DeleteTimeSeriesProcedure.java

```diff
@@ -0,0 +1,611 @@
+                .sendAsyncRequestToDataNodeWithRetry(
+                    new TConstructSchemaBlackListReq(consensusGroupIdList, patternTreeBytes),
+                    handler);
+            handler
+                .getDataNodeResponseStatusMap()
+                .forEach(
+                    (k, v) -> {
+                      if (v.getCode() == TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+                        responseMap.put(k, v);
+                      }
+                    });
+            return handler.getDataNodeResponseStatusMap();
+          }
+        };
+    constructBlackListTask.execute();
+
+    if (isFailed()) {
+      return 0;
+    }
+
+    int preDeletedNum = 0;
+    for (TSStatus resp : constructBlackListTask.getResponseMap().values()) {
+      preDeletedNum += Integer.parseInt(resp.getMessage());
+    }
+    return preDeletedNum;
+  }
+
+  private void invalidateCache(ConfigNodeProcedureEnv env) {
+    Map<Integer, TDataNodeLocation> dataNodeLocationMap =
+        env.getConfigManager().getNodeManager().getRegisteredDataNodeLocations();
+    InvalidateMatchedSchemaCacheHandler handler =
+        new InvalidateMatchedSchemaCacheHandler(dataNodeLocationMap);
+    AsyncDataNodeClientPool.getInstance()
+        .sendAsyncRequestToDataNodeWithRetry(
+            new TInvalidateMatchedSchemaCacheReq(patternTreeBytes), handler);
+    Map<Integer, TSStatus> statusMap = handler.getDataNodeResponseStatusMap();
+    for (TSStatus status : statusMap.values()) {
+      // all dataNodes must clear the related schema cache
+      if (status.getCode() != TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+        LOGGER.error("Failed to invalidate schema cache of timeseries {}", requestMessage);
+        setFailure(new ProcedureException("Invalidate schema cache failed"));
+        return;
+      }
+    }
+
+    setNextState(DeleteTimeSeriesState.DELETE_DATA);
+  }
+
+  private void deleteData(ConfigNodeProcedureEnv env) {
+    Map<TConsensusGroupId, TRegionReplicaSet> relatedSchemaRegionGroup =
+        env.getConfigManager().getRelatedSchemaRegionGroup(patternTree);
+    Map<TDataNodeLocation, List<TConsensusGroupId>> dataNodeSchemaRegionGroupGroupIdMap =
+        getLeaderDataNodeRegionGroupMap(
+            env.getConfigManager().getPartitionManager().getAllLeadership(),
+            relatedSchemaRegionGroup);
+
+    // fetch schema black list by dataNode
+    for (Map.Entry<TDataNodeLocation, List<TConsensusGroupId>> entry :
+        dataNodeSchemaRegionGroupGroupIdMap.entrySet()) {
+      Map<TConsensusGroupId, TRegionReplicaSet> targetSchemaRegionGroup = new HashMap<>();
+      entry
+          .getValue()
+          .forEach(
+              consensusGroupId ->
+                  targetSchemaRegionGroup.put(
+                      consensusGroupId, relatedSchemaRegionGroup.get(consensusGroupId)));
+      PathPatternTree patternTree =
+          fetchSchemaBlackListOnTargetDataNode(env, targetSchemaRegionGroup);
```

Comment:
Why do we fetch patternTree from black list rather than the class variable ?

Reply:
We need to resolve the pathPattern into full path representing timeseries, or at list device path based pathPattern. The storage engine need the accurate path to release memory resources, since they are organized by device path.  

Current implementation is to resolve pathPattern to accurate timeseries directly, which protect the concurrent new created timeseries from being affected by current delete timeseries task.